contrib/snap/snapcraft.yaml

```diff
@@ -0,0 +1,68 @@
+name: dogecoin-core # you probably want to 'snapcraft register <name>'
+version: '1.14.4'
+summary: Reference client of Dogecoin, a peer-to-peer digital currency like Bitcoin.
+description: |
+  Dogecoin is a cryptocurrency like Bitcoin, although it does not use SHA256 as its proof of work (POW). Taking development cues from Tenebrix and Litecoin, Dogecoin currently employs a simplified variant of scrypt. MIT licenced.
+
+base: core18
+grade: stable
+confinement: strict
+architectures:
+  - build-on: [amd64, i386, arm64]
+
+apps:
+  d:
+    command: dogecoind
+    plugs: [home, removable-media, network, network-bind]
+    environment:
+    # Override HOME so the datadir is located at ~/snap/dogecoin-core/common/.dogecoin/ instead of ~/snap/dogecoin-core/current/.dogecoin/, and each new version of the snap won't have a different data directory: https://docs.snapcraft.io/environment-variables/7983
+      HOME: $SNAP_USER_COMMON
+  qt:
+    command: qt4-launch dogecoin-qt
+    plugs: [home, removable-media, network, network-bind, desktop, x11, unity7]
+    environment:
+      HOME: $SNAP_USER_COMMON
+      DISABLE_WAYLAND: 1
+  cli:
+    command: dogecoin-cli
+    plugs: [home, removable-media, network]
+    environment:
+      HOME: $SNAP_USER_COMMON
+
+parts:
+  dogecoin-core:
+    plugin: nil
+    override-build: |
+      echo "Downloading files..."
+      wget https://github.com/dogecoin/dogecoin/releases/download/v${SNAPCRAFT_PROJECT_VERSION}/dogecoin-${SNAPCRAFT_PROJECT_VERSION}-${SNAPCRAFT_ARCH_TRIPLET}.tar.gz
+      wget https://github.com/dogecoin/gitian.sigs/archive/refs/heads/master.zip
+      unzip master.zip
+      echo "Verifying signatures..."
+      checksum=$(sha256sum dogecoin-${SNAPCRAFT_PROJECT_VERSION}-${SNAPCRAFT_ARCH_TRIPLET}.tar.gz)
+      if ! grep -r $checksum *; then
```

Comment:
Can we create a variable for the expected checksum, please, so it's a bit more obvious it will need filling in at release time?

Reply:
That said; realising this creates a chicken-egg problem in that it needs filling in after the release is tagged, doesn't it? Hmm. Probably worth calling out in the docs if we expect end users to use this.

Reply:
If we fill in the expected checksum, we would only be able to target version 1.14.3 with this. Is there a way we could make the current system more robust? Currently, it checks if the checksum generated exists anywhere within the gitian repo, which should be updated at each release.

Reply:
I think we should have the checksums in a specific file in a specific branch and not grep over the whole repository, just to close the gap that someone _could_ exploit it with a hash somewhere else?

Reply:
There are no public preimage or collision attacks on full SHA-256 right now, so that would be quite the accomplishment.

Reply:
i like it the way they did it here http://qt-mirror.dannhauer.de/archive/qt/5.15/5.15.2/submodules/md5sums.txt